lightningd/peer_control.c

```diff
@@ -657,6 +657,38 @@ static int channel_msg(struct subd *sd, const u8 *msg, const int *unused)
 	return 0;
 }
 
+struct channeld_start {
+	struct peer *peer;
+	const u8 *initmsg;
```

Comment:
Clever, you're keeping the message in its unparsed state, while in 54b4ad27b81fa7ea8ea4e2c1024bdfabe0caec08 I am parsing it and storing the individual components.